pkg/jobmanager/jobmanager.go

```diff
@@ -45,3 +55,69 @@ func (jm *JobManager) LookupJob(id uuid.UUID) (*Job, error) {
 	}
 	return j, nil
 }
+
+func (jm *JobManager) StopJob(id uuid.UUID) error {
+	j, err := jm.LookupJob(id)
+	if err != nil {
+		return err
+	}
+	j.mu.Lock() // NOTE: We don't defer unlock as we want to wait after unlock unpon success.
+	if j.status != pb.JobStatus_JOB_STATUS_RUNNING {
+		j.mu.Unlock()
+		return nil
```

Comment:
Suggestion: most manual unlocks come right before a return, suggesting we could use a defer (making the lock easier to reason about). The exception is the "j.mu.Unlock(); <-j.waitchan; return" part, but that has alternatives too.